frontend/src/metabase/ui/components/inputs/Radio/Radio.styled.tsx

```diff
@@ -14,11 +10,39 @@ export const getRadioOverrides = (): MantineThemeOverride["components"] => ({
     defaultProps: {
       size: "md",
     },
-    styles: (
-      theme: MantineTheme,
-      { labelPosition: _labelPosition }: RadioStylesParams,
-      { size = "md" },
-    ) => ({
+    variants: {
+      pill: (theme: MantineTheme) => ({
```

Comment:
Would the mantine chip component fulfill your needs here?

https://mantine.dev/core/chip/